meilisearch-http/src/routes/document.rs

```diff
@@ -66,16 +67,15 @@ async fn delete_document(
     let index = data
         .db
         .open_index(&path.index_uid)
-        .ok_or(ResponseError::index_not_found(&path.index_uid))?;
-
-    let mut update_writer = data.db.update_write_txn()?;
+        .ok_or(Error::index_not_found(&path.index_uid))?;
 
     let mut documents_deletion = index.documents_deletion();
     documents_deletion.delete_document_by_external_docid(path.document_id.clone());
 
-    let update_id = documents_deletion.finalize(&mut update_writer)?;
-
-    update_writer.commit()?;
+    let update_id = data.db.update_write::<_, _, ResponseError>(|writer| {
+        let update_id = documents_deletion.finalize(writer)?;
+        Ok(update_id)
+    })?;
```

Comment:
Is it possible to write it this way?
```suggestion
    let update_id = data.db.update_write::<_, _, ResponseError>(|writer| {
        documents_deletion.finalize(writer)
    })?;
```

Reply:
Finalize returns a heed error, and rust is not able to cast error through 2 levels of indirection, I have to wrap it in an Ok anyway, I thought readability was better this way